src/main/java/net/sf/javaanpr/intelligence/Parser.java

```diff
@@ -293,4 +294,22 @@ public void addChar(char chr) {
         }
     }
 
+    public enum SyntaxAnalysisMode {
```

Comment:
Lets pull this out into a separate file. Also, you can put the `getSyntaxAnalysisModeFromInt` method into the enum itself.